src/helm/benchmark/window_services/local_window_service.py

```diff
@@ -92,7 +92,10 @@ def truncate_from_right(self, text: str, expected_completion_token_length: int =
         max_length: int = self.max_request_length - expected_completion_token_length
         result: str = self.decode(self.encode(text, truncation=True, max_length=max_length).tokens)
 
-        # Validate that the truncated text now fits. Fail fast otherwise.
-        num_tokens: int = self.get_num_tokens(result)
-        assert num_tokens <= max_length, f"Truncation failed ({num_tokens} > {max_length}). Input text: {text}"
+        # HACK: For the vast majority of cases, the above logic works, but it sometimes doesn't work
+        # for non-English, non-Chinese text (e.g., Russian from multi_eurlex. See more
+        # in https://github.com/stanford-crfm/helm/issues/1448).
+        # Truncate by removing character by character until the prompt fits within the context window.
+        while not self.fits_within_context_window(result, expected_completion_token_length):
+            result = result[:-1]
```

Comment:
Naively truncating characters from the right until it fits the context window sounds reasonable to me.

cc @teetone @percyliang FYI for the change in window service behavior.

Reply:
I think truncation in the middle may also work (This avoid truncating `"Answer: "`). Another approach is to use LLM itself to extract a shorter summary of the context.

Reply:
I revised it this way to align with the other window services.